examples/oft-hyperliquid/test/foundry/HyperLiquidComposer.t.sol

```diff
@@ -0,0 +1,183 @@
+            coreIndexId: 1231,
+            decimalDiff: 18 - 6
+        });
+
+        HYPE = IHyperAsset({
+            assetBridgeAddress: 0x2222222222222222222222222222222222222222,
+            coreIndexId: 1105,
+            decimalDiff: 18 - 8
+        });
+
+        spotBalancePrecompileMock.setSpotBalance(OFT.assetBridgeAddress, OFT.coreIndexId, type(uint64).max);
+        spotBalancePrecompileMock.setSpotBalance(HYPE.assetBridgeAddress, HYPE.coreIndexId, type(uint64).max);
+
+        oft = new MyHyperLiquidOFT("test", "test", HL_LZ_ENDPOINT_V2, msg.sender);
+        hyperLiquidComposer = new HyperLiquidComposer(
+            HL_LZ_ENDPOINT_V2,
+            address(oft),
+            OFT.coreIndexId,
+            OFT.decimalDiff
+        );
+        DST_EID = oft.endpoint().eid();
+
+        vm.deal(HL_LZ_ENDPOINT_V2, 100 ether);
+    }
+
+    function test_deployment() public view {
+        IHyperAsset memory hypeAsset = hyperLiquidComposer.getHypeAsset();
+        assertEq(hypeAsset.assetBridgeAddress, HYPE.assetBridgeAddress);
+        assertEq(hypeAsset.coreIndexId, HYPE.coreIndexId);
+        assertEq(hypeAsset.decimalDiff, HYPE.decimalDiff);
+
+        IHyperAsset memory oftAsset = hyperLiquidComposer.getOFTAsset();
+        assertEq(oftAsset.assetBridgeAddress, OFT.assetBridgeAddress);
+        assertEq(oftAsset.coreIndexId, OFT.coreIndexId);
+        assertEq(oftAsset.decimalDiff, OFT.decimalDiff);
+
+        assertEq(spotBalancePrecompileMock.balanceOf(OFT.assetBridgeAddress, OFT.coreIndexId), type(uint64).max);
+        assertEq(spotBalancePrecompileMock.balanceOf(HYPE.assetBridgeAddress, HYPE.coreIndexId), type(uint64).max);
+    }
+
+    function test_SendSpot_no_FundAddress() public {
+        bytes memory composeMsg = abi.encodePacked(userB);
+
+        // Build composerMsg similar to the outcome of OFTCore.send()
+        bytes memory composerMsg_ = OFTComposeMsgCodec.encode(
+            0,
+            SRC_EID,
+            amount,
+            abi.encodePacked(addressToBytes32(userA), composeMsg)
+        );
+
+        assertEq(oft.balanceOf(address(hyperLiquidComposer)), 0);
+
+        // Mocks the lzReceive call which mints the tokens to the hyperLiquidComposer
+        deal(address(oft), address(hyperLiquidComposer), amount);
+
+        assertEq(oft.balanceOf(address(hyperLiquidComposer)), amount);
+
+        // Expect the Transfer event to be emitted
+        vm.expectEmit(address(oft));
+        emit IERC20.Transfer(address(hyperLiquidComposer), OFT.assetBridgeAddress, amount);
+
+        // Expect the SpotSend event to be emitted
+        vm.expectEmit(hyperLiquidComposer.L1WritePrecompileAddress());
+        emit IHyperLiquidWritePrecompile.SpotSend(
+            address(hyperLiquidComposer),
+            userB,
+            OFT.coreIndexId,
+            uint64(amount / 10 ** OFT.decimalDiff)
+        );
+
+        vm.startPrank(HL_LZ_ENDPOINT_V2);
```

Comment:
nit: `vm.prank(HL_LZ_ENDPOINT_V2);` instead since its just one call

^ 5 other instances of that